File: ui/src/types/quote.ts
export interface QuoteLineItem {
  productId: string;
  baseCost: number;
  adjustedCost: number;
  finalPrice: number;
  markupPercentage: number;
  currency: string;
}

export interface Quote {
  id: string;
  itineraryId: string;
  totalCost: number;
  totalPrice: number;
  margin: number;
  currency: string;
  status: 'Draft' | 'Generated';
  createdAt: string;
  lineItems: QuoteLineItem[];
}

export interface GenerateQuoteRequest {
  itineraryId: string;
  pax: number;
  currency: string;
  markup: number;
}

File: src/AI.Forged.TourOps.Domain/Entities/Quote.cs
using AI.Forged.TourOps.Domain.Enums;

namespace AI.Forged.TourOps.Domain.Entities;

public class Quote
{
    public Guid Id { get; set; }
    public Guid ItineraryId { get; set; }
    public Guid? LeadCustomerId { get; set; }
    public decimal TotalCost { get; set; }
    public decimal TotalPrice { get; set; }
    public decimal Margin { get; set; }
    public string Currency { get; set; } = string.Empty;
    public QuoteStatus Status { get; set; }
    public DateTime CreatedAt { get; set; }

    public Itinerary Itinerary { get; set; } = null!;
    public Customer? LeadCustomer { get; set; }
    public Booking? Booking { get; set; }
    public ICollection<QuoteLineItem> LineItems { get; set; } = new List<QuoteLineItem>();
    public ICollection<Invoice> Invoices { get; set; } = new List<Invoice>();
}
